internal/controller/controllers/infraenv_controller.go

```diff
@@ -343,13 +349,23 @@ func (r *InfraEnvReconciler) ensureISO(ctx context.Context, log logrus.FieldLogg
 			}
 			clusterDeploymentRefErr := newKubeAPIError(errors.Wrapf(err, errMsg), clientError)
 
-			// Update that we failed to retrieve the clusterDeployment
-			conditionsv1.SetStatusConditionNoHeartbeat(&infraEnv.Status.Conditions, conditionsv1.Condition{
-				Type:    aiv1beta1.ImageCreatedCondition,
-				Status:  corev1.ConditionUnknown,
-				Reason:  aiv1beta1.ImageCreationErrorReason,
-				Message: aiv1beta1.ImageStateFailedToCreate + ": " + clusterDeploymentRefErr.Error(),
-			})
+			if r.ImageServiceEnabled {
+				// Update that we failed to retrieve the clusterDeployment
+				conditionsv1.SetStatusConditionNoHeartbeat(&infraEnv.Status.Conditions, conditionsv1.Condition{
+					Type:    aiv1beta1.ImageCreatedCondition,
+					Status:  corev1.ConditionUnknown,
+					Reason:  aiv1beta1.ImageCreationErrorReason,
+					Message: aiv1beta1.ImageStateFailedToCreate + ": " + clusterDeploymentRefErr.Error(),
+				})
+			}
+			if !r.ImageServiceEnabled {
+				conditionsv1.SetStatusConditionNoHeartbeat(&infraEnv.Status.Conditions, conditionsv1.Condition{
+					Type:    aiv1beta1.ClusterDeploymentReference,
+					Status:  corev1.ConditionUnknown,
+					Reason:  aiv1beta1.MissingClusterDeploymentReason,
+					Message: aiv1beta1.MissingClusterDeploymentReference + ": " + clusterDeploymentRefErr.Error(),
+				})
+			}
```

Comment:
I'm not sure about this change ... We're expressing exactly the same situation, but changing the condition to avoid saying "image"?

I think I'd rather just leave this as is for now rather than introduce a new condition.

If anything I think it would be worth it to just change the existing condition type to something generic like "Available" which is pretty common.